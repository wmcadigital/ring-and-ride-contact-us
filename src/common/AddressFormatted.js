import PropTypes from "prop-types";

const AddressFormatted = ({ addressObj }) => {
  const { addressLine1, addressLine2, townOrCity, county, postCode } =
    addressObj;
  return (
    <>
      <div>{addressLine1}</div>
      {addressLine2 ? <div>{addressLine2}</div> : undefined}
      <div>{townOrCity}</div>
      <div>{county}</div>
      <div>{postCode}</div>
    </>
  );
};

AddressFormatted.propTypes = {
  addressObj: PropTypes.shape({
    addressLine1: PropTypes.string,
    addressLine2: PropTypes.string,
    townOrCity: PropTypes.string,
    county: PropTypes.string,
    postCode: PropTypes.string,
  }),
};

export default AddressFormatted;
