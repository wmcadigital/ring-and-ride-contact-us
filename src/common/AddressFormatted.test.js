import { createRenderer } from "react-test-renderer/shallow";

import AddressFormatted from "./AddressFormatted";

const testAddress = {
  addressLine1: "20 Test St",
  townOrCity: "Birmingham",
  county: "West Midlands",
  postCode: "B17 7UJ",
};

describe("testAddress", () => {
  it("formats an address for display", () => {
    const renderer = createRenderer();

    renderer.render(<AddressFormatted addressObj={testAddress} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("formats an address for display with address line 2", () => {
    const testAddressWithLine2 = {
      ...testAddress,
      addressLine2: "additional line",
    };

    const renderer = createRenderer();

    renderer.render(<AddressFormatted addressObj={testAddressWithLine2} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
