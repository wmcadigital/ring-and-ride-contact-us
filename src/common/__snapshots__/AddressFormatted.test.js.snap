// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testAddress formats an address for display 1`] = `
<React.Fragment>
  <div>
    20 Test St
  </div>
  <div>
    Birmingham
  </div>
  <div>
    West Midlands
  </div>
  <div>
    B17 7UJ
  </div>
</React.Fragment>
`;

exports[`testAddress formats an address for display with address line 2 1`] = `
<React.Fragment>
  <div>
    20 Test St
  </div>
  <div>
    additional line
  </div>
  <div>
    Birmingham
  </div>
  <div>
    West Midlands
  </div>
  <div>
    B17 7UJ
  </div>
</React.Fragment>
`;
